src/legacy/core_plugins/kibana/public/dashboard/dashboard_app_controller.tsx

```diff
@@ -361,7 +378,15 @@ export class DashboardAppController {
 
     const refreshDashboardContainer = () => {
       const changes = getChangesFromAppStateForContainerState();
-      if (changes && dashboardContainer) {
+      if (
+        changes &&
+        dashboardContainer &&
+        !(
+          changes.filters &&
+          Object.keys(changes).length === 1 &&
+          !areFilterChangesFetchRelevant($scope.model.filters, changes.filters)
+        )
```

Comment:
nit: Might just be me, but I re-read this conditional 3 times before I understood it 😄   Probably due to the double-negation. Maybe there's a way to simplify?

Reply:
That's exactly what I meant by

> Input greatly appreciated before I clean up and add tests.

in the PR description 😄

Just wanted to make sure the approach is sound. I will ping again once this leaves draft status.

Reply:
I think having this logic here will get us into trouble.  I didn't test but I feel like things can get out of sync now.  Like, lets say you change a disabled filter from `abc` to `xyz`.  You then add a new panel into the container. This change comes from the container, so this code gets run:

```
          inputSubscription = dashboardContainer.getInput$().subscribe(async () => {
            let dirty = false;

            // This has to be first because handleDashboardContainerChanges causes
            // appState.save which will cause refreshDashboardContainer to be called.

            // Add filters modifies the object passed to it, hence the clone deep.
            if (!_.isEqual(container.getInput().filters, queryFilter.getFilters())) {
              await queryFilter.addFilters(_.cloneDeep(container.getInput().filters));

              dashboardStateManager.applyFilters($scope.model.query, container.getInput().filters);
              dirty = true;
            }
```

At this point, I think we might have a bug because it's going to look like the container updated the disabled filter back to `abc` and the `xyz` changes will get overwritten.

I think maybe the safer bet is to just let every embeddable handle when to fetch or not.  We could have a bigger discussion on whether it even makes sense to send disabled filters down to the embeddables. The logic in this class is difficult to follow with state updates coming from so many different places but I think it will get much clearer once we get rid of angular and can clean this up.

Reply:
You are right, that's exactly what happens. Is this code block handling the case when a filter is added from within the dashboard (e.g. clicking a bar in a bar chart)?

Not sending disabled filters down to the embeddables is an interesting thought, but it might be useful to have this information there in the future (another thing which isn't solved right in my code). Handling this for each embeddable individually is probably the best solution, I will adjust my PR.

Reply:
> this code block handling the case when a filter is added from within the dashboard (e.g. clicking a bar in a bar chart)?

Yes, at least for saved searches. For visualizations I think it actually goes through the angular service and by-passes the container but it shouldn't and we need to fix it to get drilldowns to work properly.  Also handles any other changes coming from a container, e.g. any actions that modify any container or embeddable state, etc.

 > Handling this for each embeddable individually is probably the best solution, I will adjust my PR.

sounds good!